docs/developer/design/dashboard-architecture.md

```diff
@@ -0,0 +1,660 @@
+    ConfigTopic -- Consumed by --> DataReduction
+
+    DataReduction -- Publishes config (workflow specs) --> ConfigTopic
+```
+
+**Key Points:**
+- Backend Services publish data streams to a single Kafka topic
+- The Dashboard consumes this data topic, feeding into internal `DataService` components
+- The Dashboard both publishes to and consumes from the config topic
+- All backend services consume the config topic for configuration updates
+- The `data_reduction` service can publish configuration messages (workflow specs, status)
+
+## High-Level Architecture
+
+```mermaid
+graph TD
+    subgraph "External Systems"
+        K1[Kafka Data Streams]
+        K2[Kafka Config Topic]
+    end
+
+    subgraph "Infrastructure Layer"
+        KB[KafkaBridge]
+        MS[MessageSource]
+    end
+
+    subgraph "Application Layer"
+        CS[ConfigService]
+        DS[DataServices]
+        PM[Param Models]
+        WC[WorkflowController]
+    end
+
+    subgraph "Presentation Layer"
+        W1[Plots]
+        UI[Workflow Widgets]
+        W2[Config Widgets]
+    end
+
+    CS <--> WC
+    DS <--> WC
+    WC --> UI
+    K1 --> MS
+    K2 <--> KB
+    MS --> DS
+    DS --> W1
+    KB --> CS
+    CS <--> PM
+    PM --> W2
+    W2 --> PM
+```
+
+### Component Overview
+
+The architecture is structured in three main layers:
+
+- **Infrastructure Layer**: Manages Kafka integration and external message sources
+- **Application Layer**: Contains business logic, orchestration, and data management
+- **Presentation Layer**: Handles GUI components and user interaction
+
+<!-- TODO: Verify if WorkflowController is implemented and how it interacts with existing components -->
+
+## Configuration Architecture - Pydantic vs Param Separation
+
+The dashboard implements a configuration system that maintains a clear separation between frontend widgets (using Param for interactive controls) and backend validation/communication (using Pydantic models throughout). The `ConfigService` operates entirely with Pydantic models, ensuring type safety and validation consistency.
+
+### Configuration Flow Overview
+
+```mermaid
+graph TB
+    subgraph "GUI Layer"
+        PW[Param Widgets]
+        PM[Param Models<br/>TOAEdgesParam]
+    end
+
+    subgraph "Configuration Service Layer"
+        CS[ConfigService<br/>Pydantic-only]
+        SV[Schema Validator<br/>Pydantic Models]
+        MB[MessageBridge<br/>KafkaBridge]
+    end
+
+    subgraph "Kafka"
+        KT[Beamlime Config Kafka Topic]
+    end
+
+    subgraph "Backend (simplified)"
+        BE[Backend Services]
+        PD[Pydantic Models<br/>TOAEdges]
+    end
+
+    PW <--> PM
+    PM -.->|"Creates Pydantic<br/>from Param state"| CS
+    CS -.->|"Validates only<br/>Pydantic models"| SV
+    CS -- JSON --> MB
+    MB  --> KT
+    KT  --> MB
+    MB -- JSON --> CS
+    CS -.->|"Callbacks with<br/>Pydantic models"| PM
+    KT --> PD
+    PD --> KT
+
+    classDef pydantic fill:#e1f5fe
+    classDef param fill:#f3e5f5
```

Comment:
This is difficult to read in dark mode (I'm viewing it on GH). Can you also adjust the text colour?